refinery/core/tests.py

```diff
@@ -2297,3 +2305,151 @@ def test_create_workflow_engine_bad_group_name(self):
+        self.dataset_b.share(public_group)
+        self.site_statistics_a = SiteStatistics.objects.create()
+
+        # Simulate a week where nothing happened
+        self.site_statistics_b = SiteStatistics.objects.create()
+
+        # Simulate another week of user activity
+        user_b = User.objects.create_user("user_b", "", "user_b")
+        user_c = User.objects.create_user("user_c", "", "user_c")
+        self.client.login(username="user_b", password="user_b")
+        self.client.login(username="user_c", password="user_c")
+        self.client.login(username="user_c", password="user_c")
+        self.client.login(username="user_c", password="user_c")
+        self.dataset_c = create_dataset_with_necessary_models(user=user_b)
+        self.dataset_d = create_dataset_with_necessary_models(user=user_b)
+        self.dataset_d.share(public_group)
+        self.dataset_e = create_dataset_with_necessary_models(user=user_c)
+        self.dataset_e.share(public_group)
+        ExtendedGroup.objects.create(name="Test Group")
+        create_tool_with_necessary_models("VISUALIZATION")  # creates a DataSet
+        create_tool_with_necessary_models("WORKFLOW")  # creates a DataSet
+        self.site_statistics_c = SiteStatistics.objects.create()
+
+    def test__get_previous_instance(self):
+        self.assertEqual(self.site_statistics_a._get_previous_instance().id, 1)
+        self.assertEqual(self.site_statistics_b._get_previous_instance(),
+                         self.site_statistics_a)
+        self.assertEqual(self.site_statistics_c._get_previous_instance(),
+                         self.site_statistics_b)
+
+    def test_datasets_uploaded(self):
+        self.assertEqual(self.site_statistics_a.datasets_uploaded, 2)
+        self.assertEqual(self.site_statistics_b.datasets_uploaded, 0)
+        self.assertEqual(self.site_statistics_c.datasets_uploaded, 5)
+
+    def test_datasets_shared(self):
+        self.assertEqual(self.site_statistics_a.datasets_shared, 1)
+        self.assertEqual(self.site_statistics_b.datasets_shared, 0)
+        self.assertEqual(self.site_statistics_c.datasets_shared, 2)
+
+    def test_users_created(self):
+        # + 2 instead of the expected 1 because the emission of
+        # the post_migrate signal creates the AnonymousUser
+        self.assertEqual(
+            self.site_statistics_a.users_created,
+            self.site_statistics_a._get_previous_instance().users_created + 2
+        )
+        self.assertEqual(self.site_statistics_b.users_created, 0)
+        self.assertEqual(
+            self.site_statistics_c.users_created,
+            self.site_statistics_c._get_previous_instance().users_created + 2
+        )
+
+    def test_groups_created(self):
+        self.assertEqual(self.site_statistics_a.groups_created, 0)
+        self.assertEqual(self.site_statistics_b.groups_created, 0)
+        self.assertEqual(self.site_statistics_c.groups_created, 1)
```

Comment:
For all of these, I might compare triples: If the first assertion failed, you'd probably be interested in what the 2nd and 3rd were doing, too.